pyaedt/filtersolutions_core/optimization_goals_table.py

```diff
@@ -0,0 +1,411 @@
+
+    def insert_row(
+        self,
+        row_index,
+        lower_frequency=None,
+        upper_frequency=None,
+        goal_value=None,
+        condition=None,
+        parameter_name=None,
+        weight=None,
+        enabled=None,
+    ):
+        """Insert a new row of parameters into the optimization goals table.
+
+        Parameters
+        ----------
+        row_index: int
+            Index of the row. Valid values range from ``0`` to ``49``, inclusive.
+        lower_frequency: str, optional
+            Lower frequency value.
+        upper_frequency: str, optional
+            Upper frequency value.
+        goal_value: str, optional
+            Goal value.
+        condition: str, optional
+            Condition value.
+        parameter_name: str, optional
+            Parameter name.
+        weight: str, optional
+            Weight value.
+        enabled: str, optional
+            Enabled value.
+        """
+        if lower_frequency:
+            lower_frequency_bytes_value = bytes(lower_frequency, "ascii")
+        else:
+            lower_frequency_bytes_value = None
+        if upper_frequency:
+            upper_frequency_bytes_value = bytes(upper_frequency, "ascii")
+        else:
+            upper_frequency_bytes_value = None
+        if goal_value:
+            goal_value_bytes_value = bytes(goal_value, "ascii")
+        else:
+            goal_value_bytes_value = None
+        if condition:
+            condition_bytes_value = bytes(condition, "ascii")
+        else:
+            condition_bytes_value = None
+        if parameter_name:
+            parameter_name_bytes_value = bytes(parameter_name, "ascii")
+        else:
+            parameter_name_bytes_value = None
+        if weight:
+            weight_bytes_value = bytes(weight, "ascii")
+        else:
+            weight_bytes_value = None
+
+        if enabled:
+            enabled_bytes_value = bytes(enabled, "ascii")
+        else:
+            enabled_bytes_value = None
+        status = self._dll.insertOptimizationGoalDefinitionRow(
+            row_index,
+            lower_frequency_bytes_value,
+            upper_frequency_bytes_value,
+            goal_value_bytes_value,
+            condition_bytes_value,
+            parameter_name_bytes_value,
+            weight_bytes_value,
+            enabled_bytes_value,
+        )
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+
+    def remove_row(self, row_index):
+        """Remove a row from the optimization goals table.
+
+        Parameters
+        ----------
+        row_index: int
+            Index of the row. Valid values range from ``0`` to ``49``, inclusive.
+        """
+        status = self._dll.removeOptimizationGoalDefinitionRow(row_index)
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+
+    def set_design_goals(self):
+        """Set the updated design goals definition."""
```

Comment:
The comment doesn't really explain more that what the name says. What does a user need to know about what this function does?